### PR TITLE
Use string converter for String type.

### LINKED ISSUE #1876: retrofit2.BuildInConverters$StringConverter let me confused
I am so confused,when send request like this

 public interface API {
       @FormUrlEncoded
       @POST("login.do?os=" + ConstantValue.OS + "&version=" + ConstantValue.VERSION)
       Observable<LoginInfo> login(@Field("cmd") String params);
  } 
the method of my custom Converter stringConverter method do not work.. (to encrypt request)
retrofit2.BuildInConvert$StringConverter is first before mine execute.. help~please~
retrofit version 2.1.0 @

### PR DESCRIPTION
**Pull request by JakeWharton**
Unlike RequestBody and ResponseBody which are library types that must be handled by default, String is a platform type and we want to offer the flexibility of doing things like using annotations to transform the contents.

Closes #1876.

### CODE REVIEW

File: retrofit/src/main/java/retrofit2/BuiltInConverters.java [hunk 1]
```diff
@@ -27,10 +27,9 @@
   public Converter<ResponseBody, ?> responseBodyConverter(Type type, Annotation[] annotations,
       Retrofit retrofit) {
     if (type == ResponseBody.class) {
-      if (Utils.isAnnotationPresent(annotations, Streaming.class)) {
-        return StreamingResponseBodyConverter.INSTANCE;
-      }
-      return BufferingResponseBodyConverter.INSTANCE;
+      return Utils.isAnnotationPresent(annotations, Streaming.class)
+          ? StreamingResponseBodyConverter.INSTANCE
+          : BufferingResponseBodyConverter.INSTANCE;
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/BuiltInConverters.java]: This is unrelated... it just looks nicer.

Comment (eneim) [inline retrofit/src/main/java/retrofit2/BuiltInConverters.java]: Did the IDE tell you to re-format this and you just accidentally pressed yes :curious_face: ?

Comment (swankjesse) [inline retrofit/src/main/java/retrofit2/BuiltInConverters.java]: 🍩
